vowpalwabbit/core/src/simple_label.cc

```diff
@@ -55,6 +55,22 @@ void VW::details::return_simple_example(VW::workspace& all, void*, VW::example&
   VW::finish_example(all, ec);
 }
 
+void VW::details::record_stats_simple_label(
+    const VW::workspace& /* all */, shared_data& sd, const VW::example& ec, VW::io::logger& /* logger */)
+{
+  const auto& ld = ec.l.simple;
+  sd.update(ec.test_only, ld.label != FLT_MAX, ec.loss, ec.weight, ec.get_num_features());
+  if (ld.label != FLT_MAX && !ec.test_only) { sd.weighted_labels += (static_cast<double>(ld.label)) * ec.weight; }
+}
+
+void VW::details::output_example_simple_label(
```

Comment:
is this function specific to the label?

Reply:
Yes, this is a general implementation for simple labels